src/js/hooks/useTablePagination.js

```diff
@@ -9,6 +9,10 @@ const useTablePagination = ({
     pageIndex: 0,
     pageSize: defaultPageSize || 5,
   });
+  // triggerFetchDataCounter triggers data fetching in useTableDataV2, preventing duplicate
+  // API calls. Previously, useEffect dependencies [filterParams, offset] caused two requests,
+  // as useTablePagination useEffect listened to filterParams and later updated offset
+  const [triggerFetchDataCounter, setTriggerFetchDataCounter] = useState(0);
```

Comment:
I don't like any ideas that are related to endless counting. Maybe you can do something like concatenating the filterParams to get the trivial value (I mean string) and then pass it as a dependency to useEffect?